haskell/src/Monocle/Backend/Queries.hs

```diff
@@ -287,6 +288,32 @@ getProjectAgg query = do
         )
       ]
 
+-- | The repos_summary query
+getTermKey :: BH.TermsResult -> Text
+getTermKey (BH.TermsResult (BH.TextValue tv) _ _) = tv
+getTermKey BH.TermsResult {} = error "Unexpected match"
+
+getTermsAgg :: BH.Query -> Text -> TenantM [BH.TermsResult]
+getTermsAgg query onTerm = do
+  search <- aggSearch query aggs
+  pure $ filter isNotEmptyTerm $ unfilteredR search
+  where
+    aggs = BH.mkAggregations "singleTermAgg" $ BH.TermsAgg $ BH.mkTermsAggregation onTerm
+    toTermsResult bucket = BH.buckets bucket
+    unfilteredR search' = maybe [] toTermsResult (BH.toTerms "singleTermAgg" search')
+    -- Terms agg returns empty terms in a buckets
+    isNotEmptyTerm :: BH.TermsResult -> Bool
+    isNotEmptyTerm tr = getTermKey tr /= ""
+
+getRepos :: TenantM [Text]
+getRepos = do
+  results <- runTermAgg
+  pure $ getRepoFullName <$> trace (show results) results
+  where
+    query = fromMaybe (error "oops") $ Q.queryBH $ Q.load Nothing mempty Nothing "repo_regex: .*"
```

Comment:
If you'd like to define query using the search language, we could look into using a QuasiQuote, or make that `fromMaybe undefined . Q.queryBH . Q.load Nothing mempty Nothing` a utility function in the Query module.

Reply:
I'll use an helper function for the moment.